app/controllers/api/v1/carts_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Api
+  module V1
+    class CartsController < Api::V1::ApiController
+      before_action :load_cart, :find_product, :cart_item_management,
+                    only: %i[add_product remove_product]
+
+      def add_product
+        @item.add_to_cart
+
+        render json: { message: response_message('added') }, status: :ok
+      end
+
+      def remove_product
+        @item.remove_from_cart
+
+        render json: { message: response_message('removed') }, status: :ok
+      end
+
+      private
+
+      def update_cart_params
+        params.require(:cart).permit(:product_id, :quantity)
+      end
+
+      def load_cart
+        @cart = current_user.carts.find_or_create_by!(status: :in_process)
+      end
+
+      def find_product
+        @product = Product.find(update_cart_params[:product_id])
+      end
```

Comment:
instead of creating before_actions you can define the method in the following way
```suggestion
      def product
        @product ||= Product.find(update_cart_params[:product_id])
      end
```
then call it as `product` like `CartItemsManagementService.new(cart, product, ...)`

Reply:
Updated using Memoization.